### PR TITLE
feat(INT-305): onboard gladly destination

### PR DESCRIPTION
**Pull request by mihir-4116**
## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/cdk/v2/destinations/gladly/utils.js
Comment (krishna2020) [inline src/cdk/v2/destinations/gladly/utils.js:1]: unit tests missing for this

File: src/services/destination/cdkV2Integration.ts [hunk 1]
```diff
@@ -126,7 +127,7 @@ export default class CDKV2DestinationService implements IntegrationDestinationSe
           metaTo.metadata = destInputArray[0].metadata;
           try {
             const doRouterTransformationResponse: RouterTransformationResponse[] =
-              await processCdkV2Workflow(destinationType, destInputArray, tags.FEATURES.ROUTER);
+              await processCdkV2Workflow(destinationType, destInputArray, tags.FEATURES.ROUTER, requestMetadata);
```
Comment (krishna2020) [inline src/services/destination/cdkV2Integration.ts:130]: Did we test other CDKv2 destinations with this contract change ?

Comment (mihir-4116) [reply src/services/destination/cdkV2Integration.ts:130]: yes, this contract change is required for new status codes and default value for this field is already added. testing is already done

File: src/cdk/v2/destinations/gladly/utils.js
Comment (mihir-4116) [reply src/cdk/v2/destinations/gladly/utils.js:1]: addressed

File: src/cdk/v2/destinations/gladly/procWorkflow.yaml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    condition: $.outputs.findCustomer.status === 400 || ($.outputs.findCustomer.status === 200 && $.outputs.findCustomer.response.length === 0) || $.getQueryParams($.context.payload) === undefined
+    template: |
+      const response = $.defaultRequestConfig()
+      response.body.JSON = $.removeUndefinedAndNullValues($.context.payload)
+      response.endpoint = $.getEndpoint(.destination)
+      response.method = "POST"
+      response.headers = $.getHeaders(.destination)
+      response
+
+  - name: updateCustomer
```
Comment (koladilip) [inline src/cdk/v2/destinations/gladly/procWorkflow.yaml:80]: combine create and update steps using else:
```yaml
- name: createCustomer
  condition: ...
  template: ...
  else:
     - name: updateCustomer

File: src/cdk/v2/destinations/gladly/utils.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+
+const getFieldValue = (field) => {
+  if (field) {
+    if (Array.isArray(field)) {
+      return field.map((item) => ({ original: item }));
+    }
+    return [{ original: field }];
+  }
+  return undefined;
```
Comment (ItsSudip) [inline src/cdk/v2/destinations/gladly/utils.js:46]: The customer can send a single email/phone as a string as well.

Comment (mihir-4116) [reply src/cdk/v2/destinations/gladly/utils.js:46]: line no 44 handles that scenario

File: src/cdk/v2/destinations/gladly/procWorkflow.yaml (hunk 2 above)
Comment (mihir-4116) [reply src/cdk/v2/destinations/gladly/procWorkflow.yaml]: addressed

File: src/cdk/v2/destinations/gladly/procWorkflow.yaml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  - name: EventType
+    path: ../../../../constants
+  - path: ./utils
+    exportAll: true
+  - name: defaultRequestConfig
+    path: ../../../../v0/util
+  - name: removeUndefinedAndNullValues
+    path: ../../../../v0/util
+  - name: getDestinationExternalID
+    path: ../../../../v0/util
```
[the remark this answers is not in the document]
Comment (koladilip) [reply src/cdk/v2/destinations/gladly/procWorkflow.yaml:11]: I can add that feature, event I felt it is required, sometime ago, but didn't implement, it is very simple to implement.

File: src/cdk/v2/destinations/gladly/utils.js (hunk 3 above)
Comment (ItsSudip) [reply src/cdk/v2/destinations/gladly/utils.js:46]: Yes my bad

**Review by ItsSudip — APPROVED** [review]

**Review by anantjain45823 — APPROVED** [review]

File: src/cdk/v2/destinations/gladly/procWorkflow.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (koladilip) [reply src/cdk/v2/destinations/gladly/procWorkflow.yaml:11]: @mihir-4116 is working on this, he will raise PR soon

**Review by koladilip — APPROVED** [review]

**Pull request merged**